### PR TITLE
feat: Add token amount input box

### PR DESCRIPTION
**Pull request by tinaszheng**
This PR creates the token amount input box for the new deposit flow.

Design reference [here](https://www.figma.com/design/fMQodZKzn9B5aZTN5G0fLJ/dYdX-%E2%80%BA-Desktop?node-id=36627-11787&node-type=canvas&t=gT6n2z2KcejahxMp-0)

https://github.com/user-attachments/assets/81f48662-aea8-4ec9-9541-6d98c3c381ad

### CODE REVIEW

File: src/views/dialogs/DepositDialog2/AmountInput.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { Icon, IconName } from '@/components/Icon';
+
+export type AmountInputProps = {
+  value: string;
+  onChange: (newValue: string) => void;
+  token: { chain: string; address: string; symbol: string };
+  onTokenClick: () => void;
+};
+
+const numericValueRegex = /^\d*(?:\\[.])?\d*$/;
```
Comment (tyleroooo) [inline src/views/dialogs/DepositDialog2/AmountInput.tsx:20]: is regex really the best way to do this? Surely we have a helper somewhere for this

File: src/constants/tokens.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { arbitrum, base, mainnet, optimism, polygon } from 'viem/chains';
+
+import { CosmosChainId } from './graz';
+
+export const USDC_ADDRESSES = {
+  [mainnet.id]: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
```
Comment (tyleroooo) [inline src/constants/tokens.ts:6]: are these safe to hard code?

**Review by tyleroooo — APPROVED** [review]

File: src/views/dialogs/DepositDialog2/AmountInput.tsx (hunk 1 above)
Comment (tinaszheng) [reply src/views/dialogs/DepositDialog2/AmountInput.tsx:20]: i think its the most efficient way :P

File: src/constants/tokens.ts (hunk 2 above)
Comment (tinaszheng) [reply src/constants/tokens.ts:6]: ya these are pretty much hard coded on circle's website too - https://developers.circle.com/stablecoins/usdc-on-main-networks

**Pull request merged**